source/common/router/router.cc

```diff
@@ -996,11 +996,21 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   const bool retry_enabled = retry_state_ && retry_state_->enabled();
   const bool redirect_enabled = route_entry_ && route_entry_->internalRedirectPolicy().enabled();
   const bool is_redirect_only = redirect_enabled && !retry_enabled;
-  const uint64_t effective_buffer_limit = calculateEffectiveBufferLimit();
```

Comment:
calculateEffectiveBufferLimit is used at lots of different position. We may need to refactor the calculateEffectiveBufferLimit to keep the consistent behavior.